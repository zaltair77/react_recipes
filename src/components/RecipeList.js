import React from 'react';
import PropTypes from 'prop-types';
import RecipeListItem from './RecipeListItem';

const RecipeList = ({ style, favorites, recipes, ...props }) => (
  <ul style={style} className="list-reset">
    {recipes.map(recipe => (
      <RecipeListItem
        key={recipe.id}
        recipe={recipe}
        favorited={favorites.includes(recipe.id)}
        {...props}
      />
    ))}
  </ul>
);

RecipeList.propTypes = {
  recipes: PropTypes.array,
  style: PropTypes.object,
  onClick: PropTypes.func,
  favorites: PropTypes.array,
  onFavorited: PropTypes.func,
};

RecipeList.defaultProps = {
  recipes: [],
  favorites: [],
};

export default RecipeList;
