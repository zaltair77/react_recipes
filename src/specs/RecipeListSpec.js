import React from 'react';
import renderer from 'react-test-renderer';
import RecipeList from '../components/RecipeList';

const testRecipes = [
  {
    id: 1,
    name: 'Recipe 1',
    category: 'Category 1',
  },
  {
    id: 2,
    name: 'Recipe 2',
    category: 'Category 2',
  },
];

describe('<RecipeList />', () => {
  test('Should not break if no recipe passed', () => {
    const component = renderer.create(<RecipeList />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Should render correctly', () => {
    const component = renderer.create(
      <RecipeList recipes={testRecipes} />,
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Should render favorite state correctly', () => {
    const component = renderer.create(
      <RecipeList recipes={testRecipes} favorites={[1]} />,
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
