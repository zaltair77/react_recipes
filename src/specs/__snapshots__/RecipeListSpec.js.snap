// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipeList /> Should not break if no recipe passed 1`] = `
<ul
  className="list-reset"
  style={undefined}
/>
`;

exports[`<RecipeList /> Should render correctly 1`] = `
<ul
  className="list-reset"
  style={undefined}
>
  <li
    className="py2 border-bottom border-bottom-dashed pointer"
    onClick={[Function]}
  >
    <span
      aria-label="favorite"
      className="mr1"
      onClick={[Function]}
      role="img"
    >
      ⬜
    </span>
    <span>
      Recipe 1
    </span>
    <span>
      Category 1
    </span>
  </li>
  <li
    className="py2 border-bottom border-bottom-dashed pointer"
    onClick={[Function]}
  >
    <span
      aria-label="favorite"
      className="mr1"
      onClick={[Function]}
      role="img"
    >
      ⬜
    </span>
    <span>
      Recipe 2
    </span>
    <span>
      Category 2
    </span>
  </li>
</ul>
`;

exports[`<RecipeList /> Should render favorite state correctly 1`] = `
<ul
  className="list-reset"
  style={undefined}
>
  <li
    className="py2 border-bottom border-bottom-dashed pointer"
    onClick={[Function]}
  >
    <span
      aria-label="favorite"
      className="mr1"
      onClick={[Function]}
      role="img"
    >
      ✅
    </span>
    <span>
      Recipe 1
    </span>
    <span>
      Category 1
    </span>
  </li>
  <li
    className="py2 border-bottom border-bottom-dashed pointer"
    onClick={[Function]}
  >
    <span
      aria-label="favorite"
      className="mr1"
      onClick={[Function]}
      role="img"
    >
      ⬜
    </span>
    <span>
      Recipe 2
    </span>
    <span>
      Category 2
    </span>
  </li>
</ul>
`;
